internal/commands/build/build.go

```diff
@@ -178,9 +183,41 @@ func buildImageUsingBuildx(app *types.App, contextPath string, opt buildOptions,
 	if err != nil {
 		return nil, err
 	}
+
+	err = fixServiceImageReferences(ctx, dockerCli, bundle, pulledImages)
+	if err != nil {
+		return nil, err
+	}
+
 	return bundle, nil
 }
 
+func fixServiceImageReferences(ctx context.Context, dockerCli command.Cli, bundle *bundle.Bundle, pulledImages []ServiceConfig) error {
+	insecureRegistries, err := internal.InsecureRegistriesFromEngine(dockerCli)
+	if err != nil {
+		return errors2.Wrapf(err, "could not retrieve insecure registries")
+	}
+	resolver := remotes.CreateResolver(dockerCli.ConfigFile(), insecureRegistries...)
+	for _, service := range pulledImages {
```

Comment:
nit: `pulledServices` instead? My first thought when I read this name is it contains image names.

Reply:
:+1: